upup/pkg/fi/nodeup/command.go

```diff
@@ -181,27 +182,40 @@ func (c *NodeUpCommand) Run(out io.Writer) error {
 
 	osTags := distribution.BuildTags()
 
-	tags := sets.NewString()
-	tags.Insert(osTags...)
-	tags.Insert(c.config.Tags...)
+	nodeTags := sets.NewString()
+	nodeTags.Insert(osTags...)
+	nodeTags.Insert(c.config.Tags...)
 
 	glog.Infof("Config tags: %v", c.config.Tags)
 	glog.Infof("OS tags: %v", osTags)
 
-	modelContext := &model.NodeupModelContext{
-		Cluster:      c.cluster,
-		Distribution: distribution,
-		Architecture: model.ArchitectureAmd64,
+	tf, err := newTemplateFunctions(c.config, c.cluster, c.instanceGroup, nodeTags)
+	if err != nil {
+		return fmt.Errorf("error initializing: %v", err)
```

Comment:
The hope is the error has the additional info.  We're initializing "stuff" here.  There's no point saying "error in newTemplateFunctions" IMO - it doesn't mean anything to the user.